### PR TITLE
Add stylesheet for the form

### PR DESCRIPTION
**Pull request by thojava**
**Description**
Implement the style for the form which firstly being used in the Newsletter subscription page.
Support style for form field's error and field hint.

**Testing**
Load the prototype and go to http://localhost:3000/lks/form

### CODE REVIEW

**Comment by smanivasag** [comment]
Hi @thojava - Thanks for the PR. Looks like the form field(s) are popping out a bit on the right side:

Screenshot 2021-03-16 at 8 53 29 AM

I understand that `nhsuk-frontend` (https://service-manual.nhs.uk/design-system/components) offers form, fields, hint, field errors, etc components, are there any issues in integrating them directly in our brXM impl. ? Thanks!

**Comment by thojava** [comment]
@smanivasag I don't know that nhsuk-frontend having such component set. I will try to integrate them.

File: app/assets/components/button/_button.scss [hunk 1]
```diff
@@ -0,0 +1,4 @@
+.nhsuk-button {
+  height: 58px;
+  padding: 0 37px 0 37px;
```
Comment (nathandavies91) [inline app/assets/components/button/_button.scss:3]: Can we use the built-in spacing classes for this please?
https://service-manual.nhs.uk/design-system/styles/spacing

The benefit being that they are responsive across mobile and desktop, and should adapt to the user's preferences (screen size, text size, etc). Fixed padding does not offer the same behaviour.

For this, I would try something like: `padding: 0 nhsuk-spacing(6);`

File: app/views/lks/form.html [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+Home - HEE
+{% endblock %}
+
+{% block content %}
+
+<form class="form" method="post" name="newsletter">
+    <h2>Subscribe to our newsletter</h2>
+
+    <div id="page0">
+        <div class="form-field form-field-error">
```
Comment (nathandavies91) [inline app/views/lks/form.html:12]: Re-use NHSUK design system components:
https://service-manual.nhs.uk/design-system/components/text-input

And also look at wrapping these sections with fieldsets:
https://service-manual.nhs.uk/design-system/components/fieldset

File: app/views/lks/form.html [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+            <input type="text" id="fullName" class="required" value="" size="40">
+        </div>
+        <div class="form-field">
+            <label for="email">Email address</label>
+            <span class="field-hint">Your business email address</span>
+            <input type="text" id="email" class="required" value="" size="40">
+        </div>
+    </div>
+
+    <button class="nhsuk-button" type="submit" name="submit">
```
Comment (nathandavies91) [inline app/views/lks/form.html:24]: Use NHSUK component here:
https://service-manual.nhs.uk/design-system/components/buttons

**Review by nathandavies91 — COMMENTED** [review]
I've made comments on the approach, but I guess to kind of second what Selva has said.. 'is there any reason for duplicating these components?'. I looks like we can re-use the ones provided by the NHSUK kit without the need to specify custom styling of HTML here.

In which case most my comments are void :)

**Comment by thojava** [comment]
@smanivasag @nathandavies91 I updated the code to just reuse the nhsuk-frontend form component. Please help to review

File: app/views/lks/form.html [hunk 4]
```diff
@@ -0,0 +1,36 @@
+{% extends "lks-layout.html" %}
+{% block pageTitle %}
+Home - HEE
+{% endblock %}
+
+{% block content %}
+
+<form class="form" method="post" name="newsletter">
```
Comment (smanivasag) [inline app/views/lks/form.html:8]: @thojava - As discussed, could we rewrite the form using `nunjucks` (like the following) for consistency ?

```
<form class="form" method="post" name="newsletter">
    <h2>Subscribe to our newsletter</h2>
    {{ input({
      "label": {
        "text": "Full name"
      },
      "id": "fullName",
      "name": "fullName",
      "autocomplete": "name",
      "errorMessage": {
        "text": "Error message goes here"
      }
    }) }}
    {{ input({
      "label": {
        "text": "Email address"
      },
      "hint": {
        "text": "Your business email address"
      },
      "id": "email",
      "name": "email",
      "autocomplete": "email"
    }) }}
    {{ button({
      "text": "Continue"
    }) }}
</form>
```

The above seem to add visually hidden `Error:` to the field error and `aria-describedby` for accessibility.

Thanks!

**Comment by thojava** [comment]
@smanivasag @nathandavies91 The template view has been rewritten using the `nunjucks`. Thank Selva for the suggestion

**Review by smanivasag — APPROVED** [review]

**Pull request merged**